pkgs/by-name/s7/s7/package.nix

```diff
@@ -0,0 +1,74 @@
+{
+  lib,
+  fetchFromGitLab,
+  stdenv,
+  notcurses,
+  nix-update-script,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "s7";
+  version = "0-unstable-2024-09-02";
+
+  src = fetchFromGitLab {
+    domain = "cm-gitlab.stanford.edu";
+    owner = "bil";
+    repo = "s7";
+    rev = "c10d3742d263276d8934af8baaa379742bd63c35";
+    hash = "sha256-cEXveNW2bl3jj21axEAC2vOzMHDH+tA1cpTcpT5NySg=";
+  };
+
+  nativeBuildInputs = [
+    notcurses
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    gcc s7.c \
+        -o s7-nrepl \
+        -O2 \
+        -I. \
+        -Wl,-export-dynamic \
+        -lm \
+        -ldl \
+        -DWITH_MAIN \
+        -DWITH_NOTCURSES \
+        -lnotcurses-core
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    cp ./s7-nrepl $out/bin/
+    ln -s $out/bin/s7-nrepl $out/bin/s7
+
+    mkdir -p $out/lib
+    cp ./s7.c ./s7.h $out/lib/
```

Comment:
Is it a good idea (from the point of view of compsed environments) to put the sources into `/lib/` as opposed to at least `/lib/s7/` ?

Does anything use the sources at runtime? Because `s7.src` already allows easy access to the sources.